gatsby-node.js

```diff
@@ -1,7 +1,163 @@
-/**
- * Implement Gatsby's Node APIs in this file.
- *
- * See: https://www.gatsbyjs.org/docs/node-apis/
- */
+const _ = require('lodash')
+const fp = require('lodash/fp')
+const path = require('path')
+const { createFilePath } = require('gatsby-source-filesystem')
+const sizeOf = require('image-size')
 
-// You can delete this file if you're not using it
+// # pure functions
+
+function getCollection(mdEdge) {
+  return mdEdge.node.fields.collection
+}
+
+// # effectful functions
+
+exports.createPages = ({ graphql, actions }) => {
+  const { createPage } = actions
+
+  const articleTemplate = path.resolve('./src/templates/article.js')
+  return graphql(
```

Comment:
There's full async/await support in node fyi. How come you chose to use promises?